src/Core/Checkout/Document/Service/DocumentMerger.php

```diff
@@ -215,7 +215,8 @@ private function createDocumentsZip(DocumentCollection $documents, Context $cont
             });
 
             $technicalName = $document->getDocumentType()?->getTechnicalName() ?? 'unknown';
-            $name = $technicalName . '_' . $document->getOrderId() . '.' . PdfRenderer::FILE_EXTENSION;
+            $documentNumber = $document->getDocumentNumber() ?? $document->getId();
+            $name = $technicalName . '_' . $document->getOrderId() . '_' . $documentNumber . '.' . PdfRenderer::FILE_EXTENSION;
```

Comment:
Small optional nitpick (the current solution already works fine like this): Isn't the `$documentNumber` already unique? I'm not sure what benefit the `$document->getOrderId()` then provides other than making the file name harder to read. There is also the "order number" which might be better suited to show to the user as a filename which order it is about 🤔